### PR TITLE
Sort `Index.difference` & `union` results for early exit scenarios

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR sorts results in `Index.difference` & `union` in the early exit scenarios similar to: https://github.com/pandas-dev/pandas/pull/51346/

On `pandas_2.0_feature_branch`:
```
= 110 failed, 101331 passed, 2091 skipped, 952 xfailed, 312 xpassed in 1064.30s (0:17:44) =
```

This PR:
```
= 87 failed, 101354 passed, 2091 skipped, 952 xfailed, 312 xpassed in 1004.34s (0:16:44) =
```

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/_base_index.py [hunk 1]
```diff
@@ -629,12 +629,18 @@ def union(self, other, sort=None):
             common_dtype = cudf.utils.dtypes.find_common_type(
                 [self.dtype, other.dtype]
             )
-            return self._get_reconciled_name_object(other).astype(common_dtype)
+            res = self._get_reconciled_name_object(other).astype(common_dtype)
+            if sort is True:
```
Comment (isVoid) [inline python/cudf/cudf/core/_base_index.py:633]: I think using `if sort` here is fine:

1. `sort` can only be `None`, `False` and `True`. `if sort` only matches `True`.
2. In the 2 shortcut paths modified here, only `sort=True` means sorting.

Comment (galipremsagar) [inline python/cudf/cudf/core/_base_index.py:633]:
```suggestion
            if sort:
```

File: python/cudf/cudf/core/_base_index.py [hunk 2]
```diff
@@ -629,12 +629,18 @@ def union(self, other, sort=None):
...
+                return res.sort_values()
+            return res
         elif not len(self):
             common_dtype = cudf.utils.dtypes.find_common_type(
                 [self.dtype, other.dtype]
             )
-            return other._get_reconciled_name_object(self).astype(common_dtype)
+            res = other._get_reconciled_name_object(self).astype(common_dtype)
+            if sort is True:
```
Comment (galipremsagar) [inline python/cudf/cudf/core/_base_index.py:641]:
```suggestion
            if sort:
```

File: python/cudf/cudf/core/_base_index.py [hunk 3]
```diff
@@ -1091,9 +1097,15 @@ def difference(self, other, sort=None):
         other = cudf.Index(other, name=getattr(other, "name", self.name))
 
         if not len(other):
-            return self._get_reconciled_name_object(other)
+            res = self._get_reconciled_name_object(other)
+            if sort is True:
```
Comment (galipremsagar) [inline python/cudf/cudf/core/_base_index.py:1101]:
```suggestion
            if sort:
```

File: python/cudf/cudf/core/_base_index.py [hunk 4]
```diff
@@ -1091,9 +1097,15 @@ def difference(self, other, sort=None):
...
+                return res.sort_values()
+            return res
         elif self.equals(other):
-            return self[:0]._get_reconciled_name_object(other)
+            res = self[:0]._get_reconciled_name_object(other)
+            if sort is True:
```
Comment (galipremsagar) [inline python/cudf/cudf/core/_base_index.py:1106]:
```suggestion
            if sort:
```

File: python/cudf/cudf/core/_base_index.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/_base_index.py]: Yup, agree. Made the change 👍

**Review by isVoid — APPROVED** [review]

**Pull request merged**